### ISSUE TITLE
persist/push: Avoid sending object properties not declared in input types

### ISSUE THREAD

**Issue by johnrix** [vuex-orm/plugin-graphql#120]
Currently when pushing an object back to my GraphQL API, I can control the name of the input type for the update request handily using the adapter method `getInputTypeName`.

However, despite returning the name of an input type that is defined in my schema, the plugin encodes all values of my object into the push request rather than just those defined in the input type.

Using the example from the documentation, if my `updatePost` mutation only allowed changes to the content and title, but not the user, my schema might look something like this:

```
type Post {
    id: ID!
    userId: ID!
    title: String
    content: String
    ....
}

input PostInput {
    title: String
    content: String
}

type Mutation {
    updatePost(id: Int!, post: PostInput!): Post
}
```

However, if I tried to `$push()` an existing post, I will get an error from the GraphQL server, since  all of the Post properties are getting encoded in the Input object, despite the input type only having title and content fields.

**Comment by cwirz** [comment]
@johnrix how bout when you are able to mark fields as read only in the model structure?
currently having the same issue and i dont want to define the fields in the backend mutation without usage.

**Comment by johnrix** [comment]
@cwirz, is there a way to flag a field as read-only within Vuex-ORM or using the graphql plugin? That would likely be one way to get around the problem if so. I can't find it right now if it exists though.

Currently, my only option is to use a [model-related custom mutation](https://vuex-orm.github.io/plugin-graphql/guide/custom-queries.html#model-related-custom-mutation).

**Comment by cwirz** [comment]
@johnrix ok ill try to implement that when i have time

**Comment by johnrix** [comment]
Sorry, I thought you were suggesting there was some way to do that already. 

As I said, it would be one way around the issue. It seems sub-optimal to me though. Is it not possible simply to only include fields that are defined in the mutation Input type? This would then require no specific effort to flag read-only fields within the client-side model definitions.

**Comment by NickSeger** [comment]
Don't mean to bump this old thread, was this ever looked into @phortx or is there a way in the current configuration that allows for selecting which model fields are included in a mutation?

**Comment by phortx** [comment]
That's a good point. I don't think this is possible currently. Maybe I can take a look in the future Feel free to create a Pull Request :)

**Comment by phortx** [comment]
Good News:

This is an issue in my current project too. The best reason to tackle that ;)

I'll provide a patch this week.

**Comment by phortx** [comment]
This should be fixed in version [1.0.0-rc.39](https://github.com/vuex-orm/plugin-graphql/releases/tag/v1.0.0-rc.39).
